chapter_attention-mechanisms/attention_vn.md

```diff
@@ -29,7 +35,14 @@ $(\mathbf{k}_1, \mathbf{v}_1), \ldots, (\mathbf{k}_n, \mathbf{v}_n)$, with $\mat
 Given a query $\mathbf{q} \in \mathbb R^{d_q}$, the attention layer returns an output $\mathbf{o} \in \mathbb R^{d_v}$ with the same shape as the value.
 -->
 
-*dịch đoạn phía trên*
+Cơ chế *tập trung* có thể coi là khái quát hóa của phép gộp với hiệu chỉnh độ chệch trên từng đầu vào.
+Tâm điểm của cơ chế tập trung là tầng tập trung.
+Vector đầu vào của tầng tập trung được gọi ngắn gọn là *truy vấn* (*query*).
+Với mỗi truy vấn, tầng tập trung trả về đầu ra dựa trên bộ nhớ là tập các cặp vector khoá-giá trị được mã hoá trong tầng tập trung này. 
```

Comment:
```suggestion
Với mỗi câu truy vấn, tầng tập trung trả về đầu ra dựa trên bộ nhớ là tập các cặp vector khoá-giá trị được mã hoá trong tầng tập trung này. 
```

Reply:
```suggestion
Với mỗi vector truy vấn, tầng tập trung trả về đầu ra dựa trên bộ nhớ là tập các cặp vector khoá-giá trị được mã hoá trong tầng tập trung này. 
```
Em nghĩ nên dịch thống nhất `query` là `vector truy vấn`. `câu truy vấn` nghe dễ hiểu sang SQL.

Reply:
Từ "câu truy vấn" mình nghĩ k hợp lắm, vì vector truy vấn ở đây đơn giản là representation của 1 token, hay 1 element nào đó trong chuỗi vào. Câu truy vấn thân thuộc với database nhưng trong bối cảnh này lại gây hiểu nhầm là vector đại diện cho cả chuỗi (câu)? @lkhphuc giải thích chút được k do thấy Phúc cũng có suggestion tương tự ở phần Transformer :3

Reply:
Tag @cuongvng nữa vì cũng có cùng ý kiến như @lkhphuc :3

Reply:
query vừa là danh từ vừa là động từ, nhưng `truy vấn` thì chỉ là động từ thôi đúng không?
Nếu vậy chỗ nào query xài như danh từ thì mình thêm các từ bổ sung thành `câu / vector /giá trị truy vấn`.
Chỗ này mình thấy ngữ cảnh cũng khá rõ rồi, nhưng nếu @davidnvq sợ nhầm sang giá trị của cả câu thì mình có thể sửa lại thành vector. 
```suggestion
Với mỗi vector truy vấn, tầng tập trung trả về đầu ra dựa trên bộ nhớ là tập các cặp vector khoá-giá trị được mã hoá trong tầng tập trung này. 
```

Reply:
em cũng đồng ý là `câu truy vấn` nghe dễ ra SQL. Vì query có thể là vector và matrix nên em nghĩ có thể tổng quát thành `mảng - array` để dịch danh từ. Mn nghĩ sao?
```suggestion
Với mỗi mảng truy vấn, tầng tập trung trả về đầu ra dựa trên bộ nhớ là tập các cặp vector khoá-giá trị được mã hoá trong tầng tập trung này. 
```

Reply:
@cuongvng query là vector Cường ơi, a chưa biết thấy bài báo nào mà query là matrix?

Reply:
> query vừa là danh từ vừa là động từ, nhưng `truy vấn` thì chỉ là động từ thôi đúng không?
> Nếu vậy chỗ nào query xài như danh từ thì mình thêm các từ bổ sung thành `câu / vector /giá trị truy vấn`.

Google qua thì thấy `truy vấn` người ta cũng được dùng (và tạm hiểu) là danh từ trong database. Thêm `vector` vào thì hợp lý nhất.

Reply:
@ngcthuong anh Thướng có gợi ý gì không ạ? hay không cần thêm từ gì trc từ `truy vấn` nữa hả anh?

Reply:
@davidnvq query là matrix ở ngay phần dưới anh
![image](https://user-images.githubusercontent.com/30251584/84893989-568a2d80-b0ca-11ea-987d-fdb41e751b21.png)

Reply:
Ngay dưới câu đang thắc mắc trên, họ ghi rõ query là vector mà:
![image](https://user-images.githubusercontent.com/30741785/84896074-411efe80-b0de-11ea-8caa-cb158e6204eb.png)
![image](https://user-images.githubusercontent.com/30741785/84896139-601d9080-b0de-11ea-9522-a0d0cdf6ed5d.png)

Q là mảng các queries q, ngta gộp lại thành matrix để thực hiện matrix-multiplcation nhanh hơn. Chứ bản chất là mình lấy mỗi query (trong Q) để tính dot product với từng keys.
![image](https://user-images.githubusercontent.com/30741785/84896214-85aa9a00-b0de-11ea-813f-dd362e14b7d4.png)

Có thể a đang hiểu nhầm ý của Cường @@

Reply:
trong đoạn này ghi rõ `q \in R^d` thì em không ý kiến gì cách dịch `vector truy vấn`, em chỉ nghĩ nếu có thể tổng quát hoá cho cả trường hợp phía sau, `Q \in R{m x d}`, thì tốt nhất, nên mới đề xuất `mảng truy vấn`.

Reply:
Yeah, trong câu này ngta đã ghi rõ là for a query thì mình nên dịch trong ngữ cảnh của câu và quanh đó thôi chứ đỡ mất công khái quát hoá hộ tác giả :3

Reply:
Nếu `truy vấn` cũng được xài như danh từ thì không sao, để im hay thêm `câu` hay `vector` đằng trước cũng được.